cloud/amazon/ec2_addresses.py

```diff
@@ -0,0 +1,409 @@
+
+
+def unassign_secondary_private_addresses(client, module, private_ips):
+    params = dict()
+    params['NetworkInterfaceId'] = module.params.get('interface_id')
+    params['PrivateIpAddresses'] = private_ips
+
+    client.unassign_private_ip_addresses(**params)
+
+
+def remove_eip_to_private_addresses(client, module):
+    params = dict()
+    changed = False
+    if module.params.get('public_ip'):
+        public_ip = module.params.get('public_ip')
+        eip_details = get_eip_details(client, module, public_ip)
+        params['AssociationId'] = eip_details['AssociationId']
+        client.disassociate_address(**params)
+        result = release_eip_address(client, module, eip_details['AllocationId'])
+        changed = True
+    elif module.params.get('interface_id'):
+        elastic_ips = get_interface_eips(client, module)
+        if elastic_ips != []:
+            changed = True
+            for eip in elastic_ips:
+                eip_details = get_eip_details(client, module, eip)
+                params['AssociationId'] = eip_details['AssociationId']
+                client.disassociate_address(**params)
+                result = release_eip_address(client, module, eip_details['AllocationId'])
+
+    return changed, result
+
+
+def release_eip_address(client, module, allocation_id):
+    params = dict()
+    if module.params.get('release_on_disassociation'):
+        params['AllocationId'] = allocation_id
+        client.release_address(**params)
+
+
+def get_interface_eips(client, module):
+    params = dict()
+    if module.params.get('interface_id'):
+        interface_details = get_interface_details(client, module)
+        all_public_ips = [n['Association'] for n in
+            interface_details['NetworkInterfaces'][0]['PrivateIpAddresses']
+                 if 'Association' in n]
+        elastic_ips = [n['PublicIp'] for n in all_public_ips if 'AllocationId' in n]
+    else:
+        module.fail_json(msg='interface_id is required.')
+
+    return elastic_ips
+
+
+def return_secondary_address_details(client, module):
+    interface_details = get_interface_details(client, module)
+    existing_secondaries = \
+        filter(lambda i:
+            i['Primary'] == False,
+            interface_details['NetworkInterfaces'][0]['PrivateIpAddresses'])
+    return existing_secondaries
+
+
+def remove_secondary_private_addresses(client, module):
+    params = dict()
+    changed = False
+    if not module.params.get('interface_id'):
+        module.fail_json(msg='interface_id is required')
+
+    if module.params.get('private_ip'):
+        private_ips = [module.params.get('private_ip')]
+    else:
+        interface_details = get_interface_details(client, module)
+        existing_secondaries = \
+            filter(lambda i:
+                i['Primary'] == False,
+                interface_details['NetworkInterfaces'][0]['PrivateIpAddresses'])
+        private_ips = [n['PrivateIpAddress'] for n in existing_secondaries]
```

Comment:
Don't need the intermediate `existing_secondaries`. `filter` isn't very pythonic these days compared to list comprehensions (and Ansible's coding standards say list comprehensions should be kept simple if possible). I think the following expresses the same functionality while keeping it fairly simple:
```
private_ips = [i['NetworkInterfaces'][0]['PrivateIpAddresses']['PrivateIpAddress'] for i in interface_details if not i['Primary']
```